singer_encodings/schema.py

```diff
@@ -0,0 +1,142 @@
+import singer
+LOGGER = singer.get_logger()
+
+def generate_schema(samples, table_spec):
+    """Function to generate the schema as the records"""
+    counts = {}
+    for sample in samples:
+        # {'id': {'integer': 45}, 'name': {'string' : 45}}
+        counts = count_sample(sample, counts, table_spec)
+
+    for key, value in counts.items():
```

Comment:
Add detailed comments for each condition to explain the scenario better.

Reply:
Added comments.